File: src/spa-aft/wp-load.js
function* wp_load(x)
{
 check(is_str,x)
 
 const result=arr()
 
 let next=true
 let page=1
 
 while(next)
 {
  const xhr=new XMLHttpRequest
  
  let done=false

  function on_load()
  {
   const response=xhr.response   
   const a=json_decode(response)

   append(result,a)
      
   let count=xhr.getResponseHeader("x-wp-totalpages")
   
   count=to_uint(count)
   done=true   
   
   if(same(page,count))
    next=false
  }

  const url=concat(x,"?page=",page,"&per_page=100")
  
  xhr.onload=on(on_load)  
  xhr.open("get",url)
  xhr.send()
  
  while(true)
  {
   if(done)
   {
    page=inc(page)
    
    break
   }
    
   yield
  }
 }
 
 return result
}

File: src/spa-aft/wp-load.cs
gn wp_load x
 check is_str x

 let r arr
 var next true
 var page 1

 while next
  let xhr new XMLHttpRequest
  let url concat x "?page=" page "&per_page=100"
  let pages run xhr_load url xhr
  let count xhr.getResponseHeader "x-wp-totalpages"
  let count to_uint count

  append r pages

  if same page count
   assign next false

  assign page inc page

  yield
 end

 ret r
end
